app/bundles/LeadBundle/Tests/EventListener/WebhookSubscriberTest.php

```diff
@@ -185,4 +188,28 @@ public function testOnCompanySaveAndDelete()
         $dispatcher->dispatch(LeadEvents::COMPANY_POST_SAVE, $event);
         $dispatcher->dispatch(LeadEvents::COMPANY_POST_DELETE, $event);
     }
+
+    public function testOnSegmentChange(): void
+    {
+        $mockModel  = $this->createMock(WebhookModel::class);
+
+        $mockModel->expects($this->once())
+            ->method('queueWebhooksByType')
+            ->with(
+                $this->callback(
+                    function ($type) {
+                        return LeadEvents::LEAD_LIST_CHANGE === $type;
+                    }
+                )
+            );
+
+        $webhookSubscriber = new WebhookSubscriber($mockModel);
+
+        $this->dispatcher->addSubscriber($webhookSubscriber);
+
+        $lead    = new Lead();
+        $segment = new LeadList();
+        $event   = new ListChangeEvent($lead, $segment, true);
+        $this->dispatcher->dispatch(LeadEvents::LEAD_LIST_CHANGE, $event);
+    }
```

Comment:
@kuzmany this test does not make much sense. You dispatch an event directly in the test and then assert that you dispatched the event. The only thing it tests is Symfony's dispatcher.

Could you instead write a functional test where you set the webhook config to send in background create a contact, a segment and a webhook, add the contact to the segment via the `ListModel::addLead()` method and then check that the payload is stored in the webhook queue?

Reply:
When the webhooks are configured to send by a background job then the payload isn't sent directly but stored in the `]webhook_queue` table instead waiting for a cron job to pick it up. So we can assert that the correct record in that table exists when the segment membership is changed.

Reply:
Yes, I am able to do it.

Reply:
I will check it later.